src/saving_tools.jl

```diff
@@ -225,37 +225,38 @@ function tag!(d::Dict{K,T}; gitpath = projectdir(), storepatch = true, force = f
 end
 
 """
-    keyname(d::Dict{K,T}, key) where {K<:Union{Symbol,String},T}
+    keyname(d::AbstractDict{K,T}, key) where {K<:Union{Symbol,String},T}
 
 Check the key type of `d` and convert `key` to the appropriate type.
 """
-function keyname(d::Dict{K,T}, key) where {K<:Union{Symbol,String},T}
+function keyname(d::AbstractDict{K,T}, key) where {K<:Union{Symbol,String},T}
     if K == Symbol
         return Symbol(key)
     end
     return String(key)
 end
 
 """
-    checktagtype!(d::Dict{K,T}) where {K<:Union{Symbol,String},T}
+    checktagtype!(d::AbstractDict{K,T}) where {K<:Union{Symbol,String},T}
 
 Check if the value type of `d` allows `String` and promote it to do so if not.
+Currently will accept an abstract dict, but will return a dict.
 """
-function checktagtype!(d::Dict{K,T}) where {K<:Union{Symbol,String},T}
+function checktagtype!(d::AbstractDict{K,T}) where {K<:Union{Symbol,String},T}
     if !(String <: T)
         d = Dict{K, promote_type(T, String)}(d)
```

Comment:
Shouldn't this line also be changed? Instead of `Dict` to use `typeof(d)` ?

Reply:
I mean if the value types already accept `String` it won't change to Dict, but otherwise it will? Sounds weird.

Reply:
It should either always change to Dict or never, for stability reasons.

Reply:
It should be. I will change that.

Reply:
Is there a current test case where this if statement does get triggered? I haven't been able to find one to test this correctly.

Reply:
Okay so I think this is now fixed. using only `typeof` does not work because it would still return the `{K,V}`  is still returned as well. Through searching I did find a function that does return the concrete type (e.g. Dict or OrderedDict, etc.  

The function is [`get_rawtype`](https://discourse.julialang.org/t/retrieve-the-type-of-abstractdict-without-parameters-from-a-concrete-dictionary-type/67567/4) from a discussion thread. The only thing is if `parentmodule` will still work even though `DataStructures` is not a dependency. My guess though is if you have to load `DataStructures` in yourself Julia should be aware of that module, since you can't use other DictTypes unless you imported it's parent module. 

I could not figure out a better way to do this. It passes the tests of sending in an OrderedDict to tagsave, but it does return a regular dict. Now, I'm not sure why because I can't find a place where it is explicitly converting it to a regular dict, but I do have a test in there.  Honestly I am not overly concerned with that behavior at the moment because if one really wanted to add tag save, etc and maintain an order they would probably have to overload `wsave` anyway additionally, `collect_results` does not return them in a set order anyway (similar to your thoughts on `dict_list`). 

As a sanity check though I have included a test to `checktagtype!` explicitly. All tests are currently in `stools_tests.jl` because Most of the usage of `OrderedDicts` is a specific tool and could technically be branched off into a separate test file if you think that would add clarity. 

Now the only thing remaining is the dependency of `DataStructures` is still in my [`deps`] section of the Project.toml. I could not run the test cases without it. Now I am on Julia 1.5.2 and there does seem to be a change to the way test-specific dependencies are added, but I don't think that is the issue here. I've tried just about everything. 

I have submitted a new pull-request though for review, but I think just about everything should be there beyond the `DataStructures` issue. I'd be happy to fix it if you have any ideas on how to get it to do so.